lib/connector.js

```diff
@@ -218,6 +218,62 @@ Connector.prototype.id = function(model, prop) {
   return p && p.id;
 };
 
+/**
+ * Return the database mapping name of the property of the model if it exists.
+ * Otherwise return the property name.
+ * Some connectors allow the column/field name to be customized
+ * at the model property definition level as `column`,
+ * `columnName`, or `field`. For example,
+ *
+ * ```json
+ * "name": {
+ *   "type": "string",
+ *   "mysql": {
+ *     "column": "NAME"
+ *   }
+ * }
+ * ```
+ * @param {String} model The target model name
+ * @param {String} prop The property name
+ *
+ * @returns {String} The database mapping name of the property of the model if it exists
+ */
+Connector.prototype.propertyMapping = Connector.prototype.column =
+function(model, property) {
+  const prop = this.getPropertyDefinition(model, property);
+  let mappingName;
+  if (prop && prop[this.name]) {
+    mappingName = prop[this.name].column || prop[this.name].columnName ||
+    prop[this.name].field || prop[this.name].fieldName;
+    if (mappingName) {
+      // Explicit column name, return as-is
+      return mappingName;
+    }
+  }
+
+  // Check if name attribute provided for column name
+  if (prop && prop.name) {
+    return prop.name;
+  }
+  mappingName = property;
+  if (typeof this.dbName === 'function') {
+    mappingName = this.dbName(mappingName);
+  }
+  return mappingName;
+};
+
+/**
+ * Return the database mapping name of the id property of the model if it exists.
+ * Otherwise return the name of the id property.
+ * @param {String} model The target model name
+ * @param {String} prop The property name
+ * @returns {String} the database mapping name of the id property of the model if it exists.
+ */
+Connector.prototype.idMapping = Connector.prototype.idColumn = function(model) {
```

Comment:
+1 to keep `idColumn` for backwards compatibility.

Depending on how we resolve my comment above and what name we choose for `column`, we may want to rename `idMapping` similarly too.

Few examples:

- `getIdPropertyName`
- `getIdName` (don't we already have this in some connectors?)
- `getIdPropertyKey`
- `mapIdPropertyName`
- `mapIdToDbField`
- `mapIdToDbFieldName`
- `getIdDbName`